src/typechecker/lisp-type.lisp

```diff
@@ -49,6 +84,40 @@ USE-FUNCTION-ENTRIES specifies whether to emit FUNCTION-ENTRY for functions, emi
       ((function-type-p ty)
        'function-entry)
 
+      ((typep (tapp-from ty) 'tycon)
+       (let ((from (tycon-name (tapp-from ty)))
+             (to   (tapp-to ty)))
+         (cond
+           ;; First we deal with specialized parametric types.
+           ;;
+           ;; (Complex :t)
+           ((and (not (null (complex-tycon)))
```

Comment:
I would either move the `find-symbol` business into here or have the function be something like `complex-type-p` which does this full check as a single predicate.